src/ch/ch_domain.c

```diff
@@ -40,6 +44,41 @@ VIR_ENUM_IMPL(virCHDomainJob,
 
 VIR_LOG_INIT("ch.ch_domain");
 
+struct _chDomainLogContext {
+    GObject parent;
+
+    int writefd;
+    int readfd; /* Only used if manager == NULL */
+    off_t pos;
+    ino_t inode; /* Only used if manager != NULL */
+    char *path;
+    virLogManagerPtr manager;
+};
+static void chDomainLogContextFinalize(GObject *obj);
+static void ch_domain_log_context_init(chDomainLogContext *logctxt G_GNUC_UNUSED)
+{
+}
+static void ch_domain_log_context_class_init(chDomainLogContextClass *klass)
+{
+    GObjectClass *obj = G_OBJECT_CLASS(klass);
+
+    obj->finalize = chDomainLogContextFinalize;
+}
+
+G_DEFINE_TYPE(chDomainLogContext, ch_domain_log_context, G_TYPE_OBJECT);
+//static void chDomainLogContextFinalize(GObject *obj);
+static void
+chDomainLogContextFinalize(GObject *object)
+{
+    chDomainLogContextPtr ctxt = CH_DOMAIN_LOG_CONTEXT(object);
+    VIR_DEBUG("ctxt=%p", ctxt);
+
+    virLogManagerFree(ctxt->manager);
+    VIR_FREE(ctxt->path);
```

Comment:
This should use g_free, the VIR_FORCE_CLOSE are still in use upstream though.

Reply:
Fixed.